packages/lab/src/drawer/Drawer.tsx

```diff
@@ -59,52 +68,69 @@ export const Drawer = forwardRef<HTMLDivElement, DrawerProps>(function Drawer(
   });
 
   const [showComponent, setShowComponent] = useState(false);
+  const { Component: FloatingComponent } = useFloatingComponent();
 
-  const { floating, context } = useDrawer({
+  const id = useId(idProp);
+
+  const { context, floating } = useFloatingUI({
     open,
     onOpenChange,
   });
 
-  const floatingRef = useForkRef<HTMLDivElement>(floating, ref);
+  const { getFloatingProps } = useInteractions([
+    useRole(context, { role: "dialog" }),
+    useClick(context),
+    useDismiss(context),
+  ]);
+
+  const handleRef = useForkRef<HTMLDivElement>(floating, ref);
+
+  const getDrawerProps = () => {
+    return getFloatingProps({
+      ref: floating,
+      id: `${id}-drawer`,
+    });
+  };
 
   useEffect(() => {
     if (open && !showComponent) {
       setShowComponent(true);
     }
-  }, [open, showComponent]);
+  }, [open, showComponent, setShowComponent]);
+
+  if (!showComponent) return <></>;
 
   return (
-    <FloatingPortal>
-      {/* The provider is needed to support the use case where an app has nested modes. The element that is portalled needs to have the same style as the current scope */}
-      <SaltProvider>
-        {showComponent && (
-          <FloatingOverlay className={withBaseName("overlay")} lockScroll>
-            <FloatingFocusManager context={context}>
-              <div
-                ref={floatingRef}
-                className={clsx(
-                  withBaseName(),
-                  withBaseName(position),
-                  {
-                    [withBaseName("enterAnimation")]: open,
-                    [withBaseName("exitAnimation")]: !open,
-                    [withBaseName(variant)]: variant,
-                  },
-                  className
-                )}
-                onAnimationEnd={() => {
-                  if (!open && showComponent) {
-                    setShowComponent(false);
-                  }
-                }}
-                {...rest}
-              >
-                {children}
-              </div>
-            </FloatingFocusManager>
-          </FloatingOverlay>
+    <Scrim>
+      <FloatingComponent
+        aria-modal="true"
+        open={open}
+        ref={handleRef}
+        aria-labelledby={`${id}-header`}
+        aria-describedby={`${id}-content`}
+        focusManagerProps={{
+          context: context,
+        }}
+        className={clsx(
+          withBaseName(),
+          withBaseName(position),
+          {
+            [withBaseName("enterAnimation")]: open,
+            [withBaseName("exitAnimation")]: !open,
+            [withBaseName(variant)]: variant,
+          },
+          className
         )}
-      </SaltProvider>
-    </FloatingPortal>
+        onAnimationEnd={() => {
+          if (!open && showComponent) {
+            setShowComponent(false);
+          }
+        }}
+        {...getDrawerProps()}
+        {...rest}
```

Comment:
Will props in `...rest` and `getDrawerProps()` ever conflict?

Reply:
swapped them around to make sure whatever is in getDrawerProps wouldn't be overridden, what other occasions could a conflict arise? 🤔

Reply:
That doesn't really solve the issues of conflict it just means the drawer functionality won't break but things could still be overridden. Does getFloatingProps not allow the extra props to be spread there?